app/src/main/java/org/simple/clinic/textInputdatepicker/TextInputDatePickerSheet.kt

```diff
@@ -71,6 +76,12 @@ class TextInputDatePickerSheet : BaseBottomSheet<
       .map(CharSequence::toString)
       .map(::YearChanged)
 
+  private fun imeDoneClicks() = listOf(dayEditText, monthEditText, yearEditText)
+      .map { it.editorActions { actionID -> actionID == EditorInfo.IME_ACTION_DONE } }
+      .toObservable()
+      .flatMap { it }
```

Comment:
Do we need to map the action keys here? Couldn't we directly map it to the `DoneClicked` instead? Like
```kt
private fun imeDoneClicks() = listOf(dayEditText, monthEditText, yearEditText)
    .map { it.editorActions { actionID -> actionID == EditorInfo.IME_ACTION_DONE } }
    .map { DoneClicked(screenKey.minDate, screenKey.maxDate) }
```

Reply:
Yes we can.

Reply:
Turns out we do need it, without it the events are not being sent to the Mobius loop.